frontends/web/src/routes/device/bitbox02/setup/wallet-restore.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { restoreFromMnemonic } from '../../../../api/bitbox02';
+import { alertUser } from '../../../../components/alert/Alert';
+import { Backup } from '../../components/backup';
+import { SetPasswordWithBackup } from './password';
+import { RestoreFromSDCardBackup } from './restore';
+import { WithSDCard } from './sdcard';
+import { Wait } from './wait';
+
+type Props = {
+  deviceID: string;
+  onAbort: () => void;
+};
+
+export const RestoreFromSDCard = ({
+  deviceID,
+  onAbort,
+}: Props) => {
+  const [status, setStatus] = useState<'restore' | 'setPassword'>('restore');
+  const [backup, setBackup] = useState<Backup>();
+
+  const onSelectBackup = (backup: Backup) => {
+    setStatus('setPassword');
+    setBackup(backup);
+  };
+
+  const onRestoreBackup = (success: boolean) => {
+    if (!success) {
+      onAbort();
```

Comment:
This is handled from BackupsV2 which should also be refactored (but not in this PR) and does only pass success but no error code.

Reply:
I quickly checked again: I think that now BackupsV2 ends up calling the parent `onAbort` which causes the component unmount so that the error message is not displayed. You can test easily removing the sd card after selecting a backup to be restored (this causes a generic error to be raised by the bb02, I assume it is the same that can happen if e.g. the sd is corrupted).

Reply:
Sure, approved :+1: